collated/test/A0148096W.md

```diff
@@ -0,0 +1,1478 @@
+                td.learnVim, td.buyBirthdayGift, td.signUpForYoga);
+        
+        assertListResult("show events", td.learnVim, 
+                td.signUpForYoga);
+
+        assertListResult("show recurring", td.learnVim);
+        
+        assertListResult("show completed");
+    }
+
+    //---------------- Utility methods ----------------------
+    
+    private void assertListResult(String command, TestTask... expectedHits ) {
+        commandBox.runCommand(command);
+        assertListSize(expectedHits.length);
+        assertResultMessage(expectedHits.length + " tasks listed!");
+        assertTrue(taskListPanel.isListMatching(expectedHits));
+    }
+}
+```
+###### \java\guitests\UndoCommandTest.java
+``` java
+
+package guitests;
+
+import org.junit.Test;
+
+import static org.junit.Assert.assertTrue;
+
+import teamfour.tasc.logic.commands.UndoCommand;
+import teamfour.tasc.testutil.TestTask;
+
+public class UndoCommandTest extends AddressBookGuiTest {
+    
+    public void prepare() {
+        commandBox.runCommand("add \"undo test case 1\"");
+        commandBox.runCommand("add \"undo test case 2\"");
+        commandBox.runCommand("add \"undo test case 3\"");
+    }
+    
+    /* Equivalence Partitions:
+     * - undo (no argument)
+     *     - undo 1 if has available history
+     *     - unchanged if no available history
+     *     
+     * - undo (positive integer)
+     *     - undo all history if more than available history
+     *     - undo count if less than or equal to available history
+     *     - unchanged if no available history
+     *     
+     * - undo (illegal arguments)
+     *     - nonpositive integer: command fails, error message shown
+     *     - not an integer: command fails, error message shown
+     *     
+     * Boundary value analysis:
+     *     - For 0 history: test with 0, 1
+     *     - For 3 history: test with 3, 4
+     */
+    
+    //---------------- undo (no argument) ----------------------
+    
+    @Test
+    public void undo_noArg_hasThreeHistory_undoOne() {
+        commandBox.runCommand("delete 1");
+        commandBox.runCommand("delete 1");
+        commandBox.runCommand("delete 1");
+        assertUndoResult("undo", createUndoSuccessResultMessage(1), 
+                td.developerMeeting,
+                td.researchWhales, td.learnVim, 
+                td.buyBirthdayGift, td.signUpForYoga);
+    }
+    
+    @Test
+    public void undo_noArg_hasZeroHistory_remainsUnchanged() {
+        assertUndoResult("undo", UndoCommand.MESSAGE_NO_PAST_COMMAND_TO_UNDO,
+                td.submitPrototype, 
+                td.submitProgressReport, td.developerMeeting,
+                td.researchWhales, td.learnVim, 
+                td.buyBirthdayGift, td.signUpForYoga);
+    }
+    
+    //---------------- undo (positive integer) ----------------------
+    
+    @Test
+    public void undo_four_hasThreeHistory_undoAll() {
+        commandBox.runCommand("delete 1");
```

Comment:
you can probably use a setup method and move these lines there; notice that they are duplicated in many tests.